hypertension/vro-main/poetry.lock

```diff
@@ -64,20 +64,20 @@ uvloop = ["uvloop (>=0.15.2)"]
 
 [[package]]
 name = "boto3"
-version = "1.17.90"
```

Comment:
This is great. Love the idea of keeping packages up to date.

Topic for an in-person VRO engineers discussion:

Do we want to play this by ear (without structure, just doing it) like here, or do we feel we need to add some sort of structure or practice around this. I bet we're fine with no structure for now. Just figure it's worth bringing up.

Reply:
I'm glad you brought that up! I'm used to having a more strict process around library updates. Mainly, we didn't set less-trusted libraries' optimistic locking to minor versions (so that the minor versions wouldn't auto-upgrade to the latest unless we explicitly wanted them to). Also we didn't update the entire lock file when we added or changed one library version. We'd specifically update or install that one library to avoid updating all versions. That likely saved us some production incident headaches, but I don't see the harm in being less strict about it at this point in our project maturity.

Have you had a more strict process around versioning/updates like that? I do think we could revisit this question once we're in some kind of staging environment.